backend/src/utils/resourceUtils.ts

```diff
@@ -133,93 +148,157 @@ const fetchInstalledKfdefs = async (fastify: KubeFastifyInstance): Promise<KfDef
-    appDef.spec.shownOnEnabledPage = enabledAppsCM?.data[appDef.metadata.name] === 'true'; // check cm isEnabled
-    appDef.spec.isEnabled = await getIsAppEnabled(fastify, appDef);
-    if (appDef.spec.isEnabled) {
-      if (!appDef.spec.shownOnEnabledPage) {
-        changed = true;
-        appDef.spec.shownOnEnabledPage = true;
-        enabledAppsCMData[appDef.metadata.name] = 'true';
+    odhApplications.forEach((app) => {
+      if (!app.spec.featureFlag || featureFlags[app.spec.featureFlag]) {
+        applicationDefs.push(app);
       }
-    }
+    });
+  } catch (e) {
+    fastify.log.error(`Error fetching applications: ${e.response?.body?.message || e.message}`);
   }
 
-  if (changed) {
-    // write enabled apps configmap
-    const cmBody: V1ConfigMap = {
-      metadata: {
-        name: enabledAppsConfigMapName,
-        namespace: namespace,
+  return Promise.resolve(applicationDefs);
+};
+
+export const fetchApplications = async (
+  fastify: KubeFastifyInstance,
+): Promise<OdhApplication[]> => {
+  const applicationDefs = await fetchApplicationDefs(fastify);
+  const applications = [];
+
+  for (const appDef of applicationDefs) {
+    const isEnabled = await getIsAppEnabled(fastify, appDef);
+    applications.push({
+      ...appDef,
+      spec: {
+        ...appDef.spec,
+        getStartedLink: getRouteForClusterId(fastify, appDef.spec.getStartedLink),
+        isEnabled,
+        link: isEnabled ? await getRouteForApplication(fastify, appDef) : undefined,
       },
-      data: enabledAppsCMData,
-    };
-    if (!enabledAppsCM) {
-      await coreV1Api.createNamespacedConfigMap(namespace, cmBody);
-    } else {
-      cmBody.data = { ...enabledAppsCM.data, ...enabledAppsCMData };
-      await coreV1Api.replaceNamespacedConfigMap(enabledAppsConfigMapName, namespace, cmBody);
-    }
+    });
   }
 
-  return Promise.resolve(applicationDefs);
+  return Promise.resolve(applications);
 };
 
 const fetchDocs = async (fastify: KubeFastifyInstance): Promise<OdhDocument[]> => {
-  const normalizedPath = path.join(__dirname, '../../../data/docs');
   const docs: OdhDocument[] = [];
   const featureFlags = getComponentFeatureFlags();
   const appDefs = await fetchApplicationDefs(fastify);
+  const quickStarts = await fetchQuickStarts(fastify);
 
-  fs.readdirSync(normalizedPath).forEach((file) => {
-    if (yamlRegExp.test(file)) {
-      try {
-        const doc: OdhDocument = jsYaml.load(
-          fs.readFileSync(path.join(normalizedPath, file), 'utf8'),
-        );
-        if (doc.spec.featureFlag) {
-          if (featureFlags[doc.spec.featureFlag]) {
-            docs.push(doc);
-          }
-          return;
-        }
-        if (!doc.spec.appName || appDefs.find((def) => def.metadata.name === doc.spec.appName)) {
+  try {
+    const customObjectsApi = fastify.kube.customObjectsApi;
+    const res = await customObjectsApi.listNamespacedCustomObject(
```

Comment:
Same problem here, should fetch in a loop.

Reply:
done.